docs/development.md

```diff
@@ -2,136 +2,103 @@
 
 This guide is only relevant for those wanting to dig deeper or make changes to
 Faasm itself. If you'd just like to write and run functions, you can refer to
-the [set-up](setup.md) instructions. 
+the [set-up instructions](setup.md). 
 
-## Recommended Set-up
+## Recommended Set-up - Docker CLI
 
-The Faasm codebase is all standard C/C++ and Python, so should work on a range 
-of set-ups, however, the recommended configuration is:
-
-- Ubuntu 20.04
-- Python 3.8
-- Clang 10
-- CMake 3.13+
-- Ninja (rather than Make)
-
-## Code style
-
-Code style is checked as part of the CI build and uses the following
-configuration:
-
-- **C/C++** - [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html)
-  with the [Mozilla C++
-  style](https://firefox-source-docs.mozilla.org/code-quality/coding-style/coding_style_cpp.html)
-- **Python** - [Black](https://github.com/psf/black) with an 80 char line limit
-- **Markdown** - wrap to 80 characters where possible (not enforced)
-- **CMake** - TBD
-- **Bash** - TBD
-
-## Checking out the repo
-
-For now, many locations rely on the code being located at
-`/usr/local/code/faasm` (note the lowercase "f"), with the latter two 
-directories _owned_ by the current user. This is annoying and something we will 
-fix in future. 
-
-You can either set this directory up directly, or just symlink it.
-
-Assuming you've checked out this code somewhere, you'll need to make sure
-submodules are up to date:
+Most development can be done using the containerised CLI. To get everything set
+up, you need to run:
 
 ```bash
-git submodule update --init
-```
+# Clone this repo
+git clone https://github.com/faasm/faasm
+cd faasm
 
-## Basic local machine set-up
+# Update submodules
+git submodule update --init
 
-Most of the local set-up is scripted with Ansible, but you need to have Python 3
-and [Ansible](https://www.ansible.com/) set up in advance.
+# Start the CLI
+./bin/cli.sh
+```
 
-The easiest way to do this is as follows:
+To build and run the tests, you can then run the following inside the container:
 
 ```bash
-# Python stuff and other dependencies
-sudo apt install python3-dev python3-pip python3-venv libcairo2-dev libtinfo5
-sudo pip3 install -U pip
+# Build the tests and the codegen targets
+inv dev.cmake
+inv dev.cc tests
+inv dev.cc codegen_func
```

Comment:
Isn't this done in the `cli.dockerfile` lines 37 and 38?

Reply:
Yes, but when using the CLI for development we mount our own `build` directory which overwrites the stuff that's built here. This is so that build artifacts are kept on the local machine and don't have to be regenerated when you recreate your CLI container. 

These _are_ used when we use the CLI for running tests though.